test/e2e/common_test.go

```diff
@@ -983,13 +984,25 @@ func populateCache(podman *PodmanTestIntegration) {
 // rmAll removes the directory and its content, when running rootless we use
 // podman unshare to prevent any subuid/gid problems
 func rmAll(podmanBin string, path string) {
+	// When using overlay as root, podman leaves a stray mount behind. This
+	// leak causes remote tests to take a loooooong time, which then causes
+	// Cirrus to time out. Unmount the stray.
+	if os.Geteuid() == 0 {
+		overlayPath := path + "/root/overlay"
+		if _, err := os.Stat(overlayPath); err == nil {
+			if err = unix.Unmount(overlayPath, unix.MNT_DETACH); err != nil {
+				GinkgoWriter.Printf("Error unmounting %s: %v\n", overlayPath, err)
+			}
+		}
+	}
```

Comment:
logically speaking it would make much more sense to move this down into the else from isRootless() instead of having two times the same condition basically.

Reply:
That was pure sloppiness on my part, rushing a patch without looking at context. I apologize. I fixed that, and made one more cleanup pass on `setup_environment.sh`.